### DISCUSSION TITLE
When using PHP storage method, resulting csv file is empty

### DISCUSSION THREAD

**Discussion by taylorwwebb** [jspsych/jsPsych#2613] [Q&A] [answered]
I'm sure I am making some silly mistake here, but I can't seem to figure it out. I am trying to store data using the method proposed here https://www.jspsych.org/7.2/overview/data/#storing-data-permanently-as-a-file, and the resulting csv file is empty. Here is a MWE that reproduces the problem:

```
<!DOCTYPE html>
<html>
  <head>
    <title>My experiment</title>
    <script src="https://unpkg.com/jspsych@7.2.1"></script>
    <script src="https://unpkg.com/@jspsych/plugin-survey-text@1.1.0"></script>
    <link href="https://unpkg.com/jspsych@7.2.1/css/jspsych.css" rel="stylesheet" type="text/css" />
  </head>
  <body></body>
  <script>

  	function saveData(data){
		var xhr = new XMLHttpRequest();
		xhr.open('POST', 'write_data.php'); // 'write_data.php' is the path to the php file described above.
		xhr.setRequestHeader('Content-Type', 'application/json');
		xhr.send(JSON.stringify({filedata: data}));
	}

  	const jsPsych = initJsPsych({
		show_progress_bar: true,
		on_finish: function(){ saveData(jsPsych.data.get().csv());
		}
	})

	var timeline = []

	var survey_text_test = {
	  type: jsPsychSurveyText,
	  questions: [
	    {
	      prompt: "This is just a test. Type anything in the box provided.", 
	      columns: 5,
	      name: 'survey_text_test_trial', 
	      required: true
	    }, 
	  ],
	};
	timeline.push(survey_text_test)

	jsPsych.run(timeline);

  </script>
</html>
```

I have confirmed that `jsPsych.data.get()` actually does contain data to be saved. `write_data.php` is copied and pasted directly from the instructions in the link above. Any thoughts on what might be going wrong?

**Comment by jodeleeuw** [comment, marked as answer]
Hi @taylorwwebb,

Have you confirmed that the PHP setup on your server is in order and that the directory that the data is being written to is writeable?

**Reply by taylorwwebb** [reply]
Hi @jodeleeuw, thank you so much for the reply. I realized that I was actually accessing a cached version of the script that did not reflect my debugging. The script posted here works.

**Reply by ZifengLIU143** [reply]
Hi @jodeleeuw, actually I had the same problem. The csv file saved in my lab server is empty. Could you please explain a bit what you said "confirm that the PHP setup on your server is **in order**" and "the directory that the data is being written to is writeable"? In addition, I pasted my codes below. Did I do something wrong in my script? Or should I contact the lab engineer directly? 

Thank you in advance, 

Zifeng

```html
 // send the data to the server
   function saveData(name, data){
      var xhr = new XMLHttpRequest();
      xhr.open('POST', 'write_data.php'); // 'write_data.php' is the path to the php file described above.
      xhr.setRequestHeader('Content-Type', 'application/json');
      xhr.send(JSON.stringify({filedata: data}));
 }

    // initialize jsPsych 
    var jsPsych = initJsPsych({
    show_progress_bar: true,
    message_progress_bar: 'Progress',
    on_finish: function() {
        saveData(jsPsych.data.get().csv())
      }   
    });
```
